### ISSUE TITLE
Annotate errors

### ISSUE THREAD

**Issue by ferhatelmas** [GetStream/stream-chat-go#9]
to provide more context to the users.

[pkg/errors](https://github.com/pkg/errors) could be leveraged.

**Comment by tbarbugli** [comment]
do you have an example?

**Comment by ferhatelmas** [comment]
For example, client has an error in `Do`, the caller of `Do` just returns the error but it could do `Wrap` or `WithMessage` to signal where that error had happened.

https://github.com/cockroachdb/errors is another package that can do this job and is more future proof than `pkg/errors` comparatively (by having go1.13 helpers)

**Comment by pascaldekloe** [comment]
Why not use the fmt.Errorf `%w` directive? 😃

**Comment by ferhatelmas** [comment]
Yes, it's a good idea but I'm not sure about the target go versions. Above packages would work in old versions but `%w` directive requires 1.13 and above. 

Also, I believe 1.13 adaptation will be much slower compared to previous versions due to modules, `GOPROXY` and `GOPRIVATE`.

**Comment by Bogdan-D** [comment]
I've heard about `cocroachdb/errors` library, it looks very great, but I've never used it before. 
`pkg/errors` was awesome for that :+1:
